ukvm/ukvm.c

```diff
@@ -41,24 +41,26 @@
 #include <assert.h>
 #include <signal.h>
 #include <pthread.h>
-
-/* for net */
-#include <sys/socket.h>
-#include <linux/if_packet.h>
-#include <net/ethernet.h>
-#include <net/if.h>
-#include <linux/if_tun.h>
 #include <poll.h>
 
-
 #include "processor-flags.h"
 #include "../kernel/interrupts.h"
 #include "ukvm.h"
+#include "ukvm_modules.h"
 #include "misc.h"
 
-
-struct ukvm_blkinfo blkinfo;
-struct ukvm_netinfo netinfo;
+struct ukvm_module *modules[] = {
+#ifdef UKVM_MODULE_BLK
+    &ukvm_blk,
+#endif
+#ifdef UKVM_MODULE_NET
+    &ukvm_net,
+#endif
+#ifdef UKVM_MODULE_GDB
+    &ukvm_gdb,
+#endif
+};
+#define NUM_MODULES (sizeof(modules) / sizeof(struct ukvm_module *))
```

Comment:
This can be a NULL-terminated array of `(struct ukvm_module *)`, then we don't need the `NUM_MODULES` stuff.